dnsimple/dnsimple.go

```diff
@@ -342,11 +342,37 @@ func CheckResponse(resp *http.Response) error {
 		return nil
 	}
 
+	bodyBytes, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+	resp.Body = io.NopCloser(bytes.NewBuffer(bodyBytes))
+
 	errorResponse := &ErrorResponse{}
 	errorResponse.HTTPResponse = resp
+	err = json.NewDecoder(resp.Body).Decode(errorResponse)
 
-	err := json.NewDecoder(resp.Body).Decode(errorResponse)
+	// Handle the case where the errors field is a map of strings
 	if err != nil {
```

Comment:
What do you think about doing an early return to simplify e.g.

```go
    if err := json.NewDecoder(resp.Body).Decode(errorResponse); err == nil {
        return errorResponse
    }
```

Then we can get rid of this check and reduce the nesting by a level.

Reply:
Great idea! Made the change [here](https://github.com/dnsimple/dnsimple-go/pull/191/commits/04f7176fd91d9418e53a218244408a1f24f6dbff)